### DISCUSSION TITLE
WPA3 not working on Cubot King Kong Mini 2

### DISCUSSION THREAD

**Discussion by sebastian-de** [phhusson/treble_experimentations#2183] [Q&A] [answered]
I'm using AOSP 12.0 v402 (vanilla) on a [Cubot King Kong Mini 2](https://github.com/phhusson/treble_experimentations/wiki/Cubot-King-Kong-Mini-2), which works great. Thanks for all the work that's put into this project!

The only problem I have is that wifi is unreliable on my device. With the update to Android 12 it stopped working almost entirely, so I tried to look into this problem.

Turns out, that WPA3 doesn't seem to work on the King Kong Mini 2 (and probably more Mediatek devices?). My AP offers both WPA2 and WPA3, but somehow the negotiation seems to fail most of the time. Rarely the phone establishes a connection with WPA2 after multiple failed attempts.
If I create a separate WPA2-only network, the phone connects to that one instantly.

Does anyone else experience this?
And does anyone know, if it's possible to deactivate WPA3 on Android?

In the logs I see this entry, maybe that's related:
`02-10 09:53:12.735  1466  1565 E WifiVendorHal: getWifiLinkLayerStats_1_3_Internal(l.1168) failed {.code = ERROR_NOT_SUPPORTED, .description = }`

**Comment by phhusson** [comment, marked as answer]
WPA3 not working is expected, however the issue is indeed that Android thinks it supports WPA3.

Fix has already been given to me, it's: https://github.com/followmsi/android_device_asus_flo/commit/04ca9c855e9701deec7084e9449a80f4607be5f6
But I need to think it properly so as not to disable WPA3 for people who have working WPA3 hw

**Reply by phhusson** [reply]
pushed in https://github.com/phhusson/device_phh_treble/commit/a4d6f5c8d35b61304c8e8dae0e68d243d39e7324

**Reply by sebastian-de** [reply]
Thank you very much for your quick reply and for applying the fix!

**Reply by sebastian-de** [reply]
I updated to AOSP 12.1 v411 and connecting to my mixed WPA2/WPA3 access point works fine now. Thanks again!
